09_simulations/models.py

```diff
@@ -162,12 +173,105 @@ def train_mlp(X_train,
     return cv_pipeline
 
 
+def train_linear_csd(X_train,
+                     y_train,
+                     train_domains,
+                     seed,
+                     n_domains=None,
+                     params=None,
```

Comment:
Could you call it something more descriptive than params? It's tripping me up that there are some parameters embedded within, but others like n_folds and batch_size are their own separate variable. You could try to embed them all or you could just rename it something like "learning_params"